<?xml version="1.0"?>
<launch>
  #######################################
  # Load Camera
  ########################################
  <include file="$(find grasp_project)/launch/rs_d435_camera.launch"/>

  ########################################
  # Load UR5 URDF
  ########################################
  <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find grasp_project)/urdf/ur5_with_cam.urdf.xacro'" />

  #######################################
  # Load Param
  ########################################
  <!-- Load joint names into the parameter server -->
  <rosparam file="$(find grasp_project)/config/joint_names.yaml" command="load" />
  
  <!-- Launch GGCNN parameters -->
  <group ns="GGCNN">
    <rosparam file="$(find grasp_project)/config/ggcnn_param.yaml" />
  </group>

  #######################################
  # Load Gripper Controllers
  ########################################
  <!-- Load Robotiq Gripper Controller -->
  <rosparam file="$(find grasp_project)/controller/ur5gripper_controllers.yaml"
            command="load"/>

  <!-- <node name="gripper_controller_spawner"
        pkg="controller_manager"
        type="spawner"
        args="gripper_controller_vel"/> -->

  
  #######################################
  # Load UR5 Controllers
  # load velocity and position controllers
  ########################################
  <rosparam file="$(find grasp_project)/controller/ur5_controllers.yaml"
            command="load"/>

  <include file="$(find ur_gazebo)/launch/controller_utils.launch"/>

  <!-- LOAD CONTROLLERS AND RUN -->
  <arg name="controllers"
       default="pos_based_pos_traj_controller"/>

  <node name="ros_control_controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="false"
        output="screen"
        args="$(arg controllers)" />

  <!-- LOAD CONTROLLERS IN STOP MODE -->
  <arg name="stopped_controllers"
       default="joint_group_vel_controller"/>

  <node name="ros_control_controller_manager"
        pkg="controller_manager"
        type="controller_manager"
        respawn="false"
        output="screen"
        args="load $(arg stopped_controllers)" />
</launch>